types/schemas/prototypeApplication/index.ts

```diff
@@ -4,81 +4,109 @@ import {ApplicationData} from './data/ApplicationData';
-  HedgerowRemovalNoticeApplicationType
->;
-export type AdvertConsent = ApplicationSpecification<
-  'advertConsent',
-  AdvertConsentApplicationType
->;
-export type ComplianceConfirmation = ApplicationSpecification<
-  'complianceConfirmation',
-  ComplianceConfirmationApplicationType
->;
-// TODO: All the rest!
+/**
+ * @internal
+ * The generic base type for all application types which include a pre-assessment result
+ * Child properties may differ based on a granular application type parameter
+ */
+interface ApplicationSpecificationWithPreAssessment<T extends ApplicationType> {
+  applicationType: T;
+  data: {
+    user: UserBase;
+    applicant: Applicant<T>;
+    application: ApplicationData<T>;
+    property: Property<T>;
+    proposal: Proposal<T>;
+  };
+  preAssessment: PreAssessment;
+  responses: Responses;
+  files: File[];
+  metadata: PrototypePlanXMetadata;
+}
+
+/**
+ * @internal This list of variants should only includes submittable application types, not generic PlanX "service" prefixes like 'ldc'
+ *   See https://docs.google.com/spreadsheets/d/1FgULPemnwuwysrYGEkReYFXz3n3T7W0nWziU00taZE4/edit?gid=0#gid=0 as reference for which are 'WithPreAssessment'
+ */
+type AdvertConsent = ApplicationSpecification<'advertConsent'>;
+type ComplianceConfirmation =
+  ApplicationSpecification<'complianceConfirmation'>;
+type HedgerowRemovalNotice = ApplicationSpecification<'hedgerowRemovalNotice'>;
+type LandDrainageConsent = ApplicationSpecification<'landDrainageConsent'>;
+type LawfulDevelopmentCertificateBreachOfCondition =
+  ApplicationSpecification<'ldc.breachOfCondition'>;
+type LawfulDevelopmentCertificateExisting =
+  ApplicationSpecificationWithPreAssessment<'ldc.existing'>;
+type LawfulDevelopmentCertificateListedBuildingWorks =
+  ApplicationSpecification<'ldc.listedBuildingWorks'>;
+type LawfulDevelopmentCertificateProposed =
+  ApplicationSpecificationWithPreAssessment<'ldc.proposed'>;
+type ListedBuildingConsent = ApplicationSpecification<'listed'>;
+type PriorApprovalPart1ClassA =
+  ApplicationSpecificationWithPreAssessment<'pa.part1.classA'>;
+type PriorApprovalPart3ClassMA =
+  ApplicationSpecificationWithPreAssessment<'pa.part3.classMA'>;
+type PriorApprovalPart7ClassM =
+  ApplicationSpecificationWithPreAssessment<'pa.part7.classM'>;
+type PriorApprovalPart14ClassJ =
+  ApplicationSpecificationWithPreAssessment<'pa.part14.classJ'>;
+type PriorApprovalPart20ClassAB =
+  ApplicationSpecificationWithPreAssessment<'pa.part20.classAB'>;
+type PlanningPermissionHouseholder =
+  ApplicationSpecification<'pp.full.householder'>;
+type PlanningPermissionHouseholderRetrospective =
+  ApplicationSpecification<'pp.full.householder.retro'>;
+type PlanningPermissionMajor = ApplicationSpecification<'pp.full.major'>;
+type PlanningPermissionMinor = ApplicationSpecification<'pp.full.minor'>;
+type WorksToTreesConsent = ApplicationSpecification<'wtt.consent'>;
+type WorksToTreesNotice = ApplicationSpecification<'wtt.notice'>;
```

Comment:
Note here that each granular variant is actually one of two options now: `ApplicationSpecification` or `ApplicationSpecificationWithPreAssessment`. This allows us to make outer `preAssessment` property requried or properly omit it, rather than always optional !